contracts/src/assertionStakingPool/EdgeStakingPool.sol

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro-contracts/blob/main/LICENSE
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity ^0.8.0;
+
+import "./AbsBoldStakingPool.sol";
+import "../challengeV2/EdgeChallengeManager.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @notice Staking pool contract for creating layer zero edges.
+/// Allows users to deposit stake, create an edge once required stake amount is reached,
+/// and reclaim their stake when and if the edge is confirmed.
+/// @dev Unlike the assertion staking pool, there is no need for a function to claim the stake back into the pool.
+contract EdgeStakingPool is AbsBoldStakingPool {
+    using SafeERC20 for IERC20;
+
+    EdgeChallengeManager public immutable challengeManager;
+    bytes32 public immutable createEdgeArgsHash;
+    uint256 public immutable requiredStake;
+
+    /// @notice The provided arguments to not match createEdgeArgsHash
+    error IncorrectCreateEdgeArgs();
+
+    /// @param _challengeManager EdgeChallengeManager contract
+    /// @param createEdgeArgs Arguments to be passed into EdgeChallengeManager.createLayerZeroEdge
+    constructor(
+        address _challengeManager,
+        CreateEdgeArgs memory createEdgeArgs
+    ) AbsBoldStakingPool(EdgeChallengeManager(_challengeManager).stakeToken()) {
+        challengeManager = EdgeChallengeManager(_challengeManager);
+        createEdgeArgsHash = keccak256(abi.encode(createEdgeArgs));
+        requiredStake = challengeManager.stakeAmounts(createEdgeArgs.level);
+    }
+
+    /// @notice Create the edge. Callable only if required stake has been reached and edge has not been created yet.
+    function createEdge(CreateEdgeArgs calldata args) external {
+        if (!isCorrectCreateEdgeArgs(args)) {
+            revert IncorrectCreateEdgeArgs();
+        }
+        stakeToken.safeIncreaseAllowance(address(challengeManager), requiredStake);
+        challengeManager.createLayerZeroEdge(args);
+    }
+
+    /// @notice Check that the provided arguments match the expected arguments
+    function isCorrectCreateEdgeArgs(CreateEdgeArgs calldata args) public view returns (bool) {
+        return keccak256(abi.encode(args)) == createEdgeArgsHash;
```

Comment:
should not role a new hashing scheme, we can validate using `ChallengeEdgeLib.id()` with the edgeId returned from `createLayerZeroEdge`

Reply:
that function uses `originId` though, which we don't have immediately available in the `CreateEdgeArgs` struct

Reply:
I think it is fair to assume the creator would be able to provide that

Reply:
ok sounds good, `requiredStake` becomes a little weird then, as i assume we'd be just accepting a hash in the constructor, so we don't know which level the edge has without an additional parameter (that could end up being wrong later on)

Reply:
or should the constructor take a `ChallengeEdge`?

Reply:
i guess then it's just a bad pool and should be ignored by honest parties

Reply:
like this? https://github.com/OffchainLabs/bold/pull/622

Reply:
left some comment there, I think we can remove `requiredStake`